src/HydraAuction/OnChain/StateToken.hs

```diff
@@ -43,7 +43,7 @@ mkPolicy (EscrowAddress escrowAddressLocal, terms) redeemer ctx =
         && exactlyOneOutputToEscrow
     BurnVoucher ->
       traceIfFalse "Not exactly one Voucher burned" (txInfoMint info == voucherOnlyValue (-1))
-        && checkInterval terms VoucherExpiredStage info
+        && (checkInterval terms VoucherExpiredStage info || checkInterval terms CleanupStage info)
```

Comment:
Nice.

Reply:
I was wrong, this does not work in general case. We need to check that interval belongs to union, not that it belongs to either of intervals.

And by spec here we need only CleanupStage here, that is what Tx code uses for validity, and that is why tests does not fail.